semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -541,6 +557,42 @@ and expr_aux env ?(void = false) e_gen =
       let args = arguments env args in
       add_instr env (mk_i (CallSpecial (Some lval, special, args)) eorig);
       mk_e (Fetch lval) (related_tok tok)
+  | G.Call
+      (* This narrowly adds some calls from objects of certain classes as default
+         propagators. For now, this is a quick fix to make rule-writing easier.
+         TODO In the future, we should only use this for classes where we perform
+         a special analysis. Other propagators should be in a configurable defaults
+         file *)
+      ( {
+          e =
+            G.DotAccess
+              ( ({
+                   e =
+                     G.N
+                       (G.Id
+                         ((id, _), { id_type = { contents = Some { t; _ } }; _ }));
+                   _;
+                 } as obj),
+                _tok,
+                G.FN
+                  (G.Id
+                    ( ((("add" | "addAll") as kind), tok),
```

Comment:
Is the method called `add` or `addAll` in all cases? I think for `HashMap` it is called `put` for example.